addon/v1alpha1/types_managedclusteraddon.go

```diff
@@ -58,6 +58,18 @@ type RegistrationConfig struct {
 	//
 	// +optional
 	Subject Subject `json:"subject,omitempty"`
+
+	// certificateStatus actively tracks the status of the certificate used by the addon.
+	CertificateStatus *RegistrationConfigCertificateStatus `json:"certificateStatus,omitempty"`
```

Comment:
this field is only used for configuration of registration. I do not think we want to expose this field to addon when setting the configuration. Maybe we should move this out with the name of `RegistrationStatuses`?

Reply:
if we're moving this out, we will need an identifier field for each certificate status struct to correlate the status and registration config. i cant find such an identifier to use, otherwise we will need to duplicate the whole config. wdyt?